src/registry/v1/client.rs

```diff
@@ -1,17 +1,15 @@
 use super::data::*;
-use crate::core::WithTracing;
 use crate::openid::TokenProvider;
-use crate::{error::ClientError, openid::TokenInjector, Translator};
+use crate::util::Client;
+use crate::{error::ClientError, Translator};
 use futures::{stream, StreamExt, TryStreamExt};
-use reqwest::{Response, StatusCode};
-use serde::de::DeserializeOwned;
 use std::fmt::Debug;
 use tracing::instrument;
 use url::Url;
 
-/// A device registry client, backed by reqwest.
+/// A device registry client
 #[derive(Clone, Debug)]
-pub struct Client<TP>
+pub struct RegistryClient<TP>
```

Comment:
I would like to keep the name as just `Client`. True, `RegistryClient` would work as well, however now that mean a lot of renaming.

From the user's perspective: when only using registry stuff, one could simply use `Client`. When adding more clients, one can still use `registry::v1::Client` or `use registry::v1::Client as RegistryClient`.

So the `Registry` prefix felt a bit redundant, as in the `registry::v1` module, I would only expect "registry" clients anyway.

Reply:
Good point